packages/treat/src/builder.ts

```diff
@@ -286,14 +286,22 @@ export function styleTree<ReturnType>(
   makeStyleTree: MakeStyleTree<ReturnType>,
 ): ReturnType {
   const themedClassRefs = new Map<ClassRef, ThemeStyleMap>();
-  const startingScope = getNextScope();
+  const assignedScopes: Array<number> = [];
+
+  const getNodeScope = (scopeIndex: number) => {
+    if (typeof assignedScopes[scopeIndex] !== 'number') {
+      assignedScopes[scopeIndex] = getNextScope();
+    }
+
+    return assignedScopes[scopeIndex];
+  };
 
   const themedTrees = getThemes().map(({ tokens, themeRef }) => {
-    let scopeCount = startingScope;
+    let scopeIndex = 0;
 
     const makeStyle = (style: Style, localDebugName?: string) => {
       const localName = localDebugName || 'styleNode';
-      const classRef = getIdentName(localName, scopeCount++);
```

Comment:
Why did this approach not work? The code still looks okay to me, so I'm obviously missing something.

Reply:
The issue was that scopeCount was local to this function. We only ever called the top level getNextScope once. This meant that style trees that had multiple output nodes and we're immediately followed by another styleTree could clash.